src/main/java/com/microsoft/aad/msal4j/MSALRuntimeAuthResult.java

```diff
@@ -0,0 +1,112 @@
+package com.microsoft.aad.msal4j;//TODO: will be move to JavaMSALRuntime package, must not reference MSAL Java directly
+
+import com.sun.jna.Structure;
+import com.sun.jna.WString;
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.ptr.LongByReference;
+
+import java.util.concurrent.CompletableFuture;
+
+//FieldOrder needs to explicitly state the public fields that should be in the C structure JNA creates
+@Structure.FieldOrder({"authParamsHandle", "accountHandle", "authResultHandle", "idToken", "accessToken", "accessTokenExpirationTime", "accountId", "accountClientInfo"})
+public class MSALRuntimeAuthResult extends Structure implements MSALRuntimeLibrary.AuthResultCallbackData {
+    public LongByReference authParamsHandle;
+    public LongByReference accountHandle;
+    public LongByReference authResultHandle;
+    public MSALRuntimeUtils.StringByReference idToken = new MSALRuntimeUtils.StringByReference(new WString(""));
+    public MSALRuntimeUtils.StringByReference accessToken = new MSALRuntimeUtils.StringByReference(new WString(""));
+    public LongByReference accessTokenExpirationTime;
+    public MSALRuntimeUtils.StringByReference accountId = new MSALRuntimeUtils.StringByReference(new WString(""));//TODO: split AuthenticationResult into SignInResult (id token/account) and TokenResult (access tokens)? Or keep it simple all-in-one like MSAL Java's AuthenticationResult?
+    public MSALRuntimeUtils.StringByReference accountClientInfo = new MSALRuntimeUtils.StringByReference(new WString(""));
```

Comment:
We shouldn't have public instance fields in our APIs, if at all possible

Reply:
These fields (and the class itself) need to be public due to how JNA handles converting Java classes to C structs. Basically, JNA will ignore any field that isn't public when creating the native struct, so anything on the Java side tried to access native structs JNA is handling you'd get an error about the struct not having a field named XYZ.